assembly/runtime.ts

```diff
@@ -776,22 +787,50 @@ export class RootContext extends BaseContext {
 
   // Can be used to validate the configuration (e.g. in the control plane). Returns false if the
   // configuration is invalid.
-  validateConfiguration(configuration_sizeconfiguration_size: size_t): bool { return true; }
+  validateConfiguration(configuration_size: usize): bool {
+    log(LogLevelValues.debug, "context id: " + this.context_id.toString() + ": validateConfiguration(configuration_size:" + configuration_size.toString() + ")");
+    return true; 
+  }
+
   // Called once when the VM loads and once when each hook loads and whenever configuration changes.
   // Returns false if the configuration is invalid.
-  onConfigure(): bool { return true; }
+  onConfigure(configuration_size: usize): bool { 
+    log(LogLevelValues.debug, "context id: " + this.context_id.toString() + ": onConfigure(configuration_size: " + configuration_size.toString() + ")");
+    // Instead of making a call to the proxy, need to read the configuration from buffer..
```

Comment:
I'm not sure how to do this properly. Still trying to understand how to use the configuration_size parameter to read this information.

Reply:
I'm sorry, but I'm not sure I understood. 
Could you please elaborate a bit more on this? 
Is this related to how any data is passed from the proxy to the plugin and vice-versa?

Reply:
when data is passed to webassembly it has to be copied to the vm memory, so instead of copying the data all the time, the host runtime only passes the sizes of the buffer. 

that way if the proxy needs the data, it copies it explicitly with a dedicated method only when it is needed.
when you use a method like proxy_get_configuration, the host runtime calls back to the filters `malloc` to allocate the buffer with the correct size. so, you dont need to use the size parameter in this case.

if you example you are creating a filter that just logs the body size, you can use the size parameter for that, instead of copying the whole body.

Reply:
This leads us to the same discussion about standardizing how to read and write to the buffers when exchanging data between the proxy and the filter. However, I think that will need to be part of another PR. I will need some time to investigate that.